src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -381,6 +385,17 @@ public int compare(TypedVar o1, TypedVar o2) {
 
   }
 
+  // Due to lack of precise definition of a namespace, we look for object types that are not of any
+  // other type. This will need more refinement as
+  private boolean isLikelyNamespace(JSType type, JSType nativeObjectType) {
+    if (type.isUnionType()) {
+      UnionType ut = (UnionType) type;
+      return ut.getAlternates().contains(nativeObjectType);
```

Comment:
Shouldn't we specifically check if it's just `null|Object` or `null|undefined|Object` etc, i.e. the silly other types? I'd expect `Banana|Object` to probably not be a namespace?